peps/pep-0791.rst

```diff
@@ -0,0 +1,158 @@
+
+Their aliases in :external+py3.14:mod:`math` will be :term:`soft deprecated`.
+
+Modules functions will accept integers and objects that implement the
+:external+py3.14:meth:`~object.__index__` method which is used to convert the
+object to an integer number.
+
+Possible extensions for the new module and its scope are discussed in the
+`Open Issues <Open Issues_>`_ section.  New functions are not part of the
+proposal.
+
+
+Backwards Compatibility
+=======================
+
+As aliases in :external+py3.14:mod:`math` will be kept for an indefinite time
+(their use would be discouraged), there are no anticipated code breaks.
+
+
+Reference Implementation
+========================
+
+https://github.com/python/cpython/pull/133909
+
+
+Open Issues
+===========
+
+Module name
+-----------
+
+Chosen name seems consistent with other domain-specific mathematical module:
+:external+py3.14:mod:`cmath` (for complex numbers).
+
+There is already an ``imath`` project on PyPI, but only with two releases, with
+the most recent one four years ago.  Its repository is no longer accessible.
+The `Imath <https://github.com/AcademySoftwareFoundation/Imath>`_ C++ library
+include Python bindings with the same name.
+
+`Polling shows <https://discuss.python.org/t/91337/35>`_ ``intmath`` as another
+popular name.  The argument made was that the normal mathematical spelling of
+the imaginary unit is ``i``, which makes imath ambiguous.  It also has no conflict
+with any PyPI module.  On the other hand, ``intmath`` may be confused with
+interval math or numerical integration.
+
+Other proposed names include ``ntheory`` (like SymPy's submodule),
+``integermath`` and ``imaths``.
+
+
+Module scope and possible extensions
+------------------------------------
+
+Unless we can just provide bindings to some well supported mathematical library
+like the GMP, the module scope should be limited.  For example, no efficient primality
+testing and factorization.
+
+There are possible additions, among proposed in the initial discussion thread
+(see also issue
+`python/cpython#81313 <https://github.com/python/cpython/issues/81313>`_):
+
+* ``c_div()`` --- for integer ceiling divide, see
+  `relevant discussion thread <https://discuss.python.org/t/91269>`_.
+* ``gcdext()`` --- to solve linear `Diophantine equation <https://en.wikipedia.org/wiki/Diophantine_equation>`_ in two variables (the
+  :external+py3.14:class:`int` implementation actually includes an extended
+  Euclidean algorithm)
+* ``isqrt_rem()`` --- to return both integer square root and a remainder (if
+  integer isn't a perfect square)
+* ``ilog()`` --- integer logarithm, :external+py3.14:func:`math.log`
+  has a special handling for integer arguments.  It's unique (with respect to other module
+  functions) and not documented so far, see issue
+  `python/cpython#120950 <https://github.com/python/cpython/issues/120950>`_.
+* ``fibonacci()`` --- `Fibonacci sequence <https://en.wikipedia.org/wiki/Fibonacci_sequence>`_.
+
```

Comment:
Other proposed functions:

* `as_integer_ratio()`
* `binom(n, k)`: `factorial(n) // (factorial(k) * factorial(n-k))` (but uses more 
efficient algorithm).

```
def as_integer_ration(x):
     if hasattr(x, 'as_integer_ration'):
         return x.as_integer_ration()
     else:
         return (x.numerator, x.denominator)
```

Reply:
`binom` is `comb`, which is already added.